llvm/include/llvm/IR/Instructions.h

```diff
@@ -1836,6 +1837,16 @@ class CallInst : public CallBase {
   bool canReturnTwice() const { return hasFnAttr(Attribute::ReturnsTwice); }
   void setCanReturnTwice() { addFnAttr(Attribute::ReturnsTwice); }
 
+  bool isNonContinuableTrap() const {
+    switch (getIntrinsicID()) {
+    case Intrinsic::trap:
+    case Intrinsic::ubsantrap:
+      return !hasFnAttr("trap-func-name");
```

Comment:
It's not obvious to me why this attribute matters. Could this just be any noreturn intrinsic?

Reply:
The replacement function is not necessarily treated as `noreturn`:
```
> clang test.c --target=x86_64 -mllvm -ftrap-function=mytrap -S -o -
...
        callq   mytrap@PLT
        popq    %rbp
        retq
```

Reply:
That seems like a separate bug?

Reply:
I don't have enough information to tell if this is a bug, but it is definitely out of the scope of this patch.